File: SistemaEscritorio/src/app/core/models/CalendarEntry.js
export class CalendarEntry {
  constructor(data = {}) {
    this.id = data.id || null;
    this.userId = data.userId || '';
    this.recipeId = data.recipeId || '';
    this.recipeTitle = data.recipeTitle || '';
    this.recipeImageUrl = data.recipeImageUrl || '';
    this.scheduledDate = data.scheduledDate || null;
    this.mealType = data.mealType || 'lunch';
    this.notificationSent = data.notificationSent || false;
    this.createdAt = data.createdAt || null;
    this.updatedAt = data.updatedAt || null;
  }

  get mealTypeLabel() {
    const types = {
      breakfast: 'Desayuno',
      lunch: 'Almuerzo',
      dinner: 'Cena',
      snack: 'Merienda'
    };
    return types[this.mealType] || this.mealType;
  }

  get isPast() {
    return this.scheduledDate && new Date(this.scheduledDate) < new Date();
  }

  get isToday() {
    if (!this.scheduledDate) return false;
    const today = new Date();
    const date = new Date(this.scheduledDate);
    return date.toDateString() === today.toDateString();
  }

  toJSON() {
    return {
      id: this.id,
      userId: this.userId,
      recipeId: this.recipeId,
      recipeTitle: this.recipeTitle,
      recipeImageUrl: this.recipeImageUrl,
      scheduledDate: this.scheduledDate,
      mealType: this.mealType,
      notificationSent: this.notificationSent,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt
    };
  }
}

File: SistemaMovil/lib/domain/entities/calendar_entry.dart
class CalendarEntry {
  String id;
  String userId;
  String recipeId;
  String recipeTitle;
  String recipeImageUrl;
  DateTime scheduledDate;
  String mealType; // 'breakfast', 'lunch', 'dinner', 'snack'
  bool notificationSent;
  DateTime createdAt;

  CalendarEntry({
    required this.id,
    required this.userId,
    required this.recipeId,
    required this.recipeTitle,
    required this.recipeImageUrl,
    required this.scheduledDate,
    required this.mealType,
    this.notificationSent = false,
    required this.createdAt,
  });
}
